logical/system_view.go

```diff
@@ -12,11 +12,16 @@ type SystemView interface {
 	// authors should take care not to issue credentials that last longer than
 	// this value, as Vault will revoke them
 	MaxLeaseTTL() time.Duration
+
+	// SudoPrivilege returns true if given path has sudo privileges under the given
+	// list of policies
+	SudoPrivilege(path string, policies []string) bool
```

Comment:
I think this is a useful function, but I see one problem: backends may not know the policies associated with the token, and they may not have a pointer into core in order to look it up, whereas DynamicSystemView does. So I'd change this to instead be given a token value (e.g. so in most requests you'd use req.ClientToken). Then look up the policies.

This does mean copying more a bit more code from core.checkToken, but I think it's worth it to make this more generally useful outside TokenStore.